coderd/database/dbauthz/dbauthz_test.go

```diff
@@ -155,3 +158,1143 @@ func must[T any](value T, err error) T {
 	}
 	return value
 }
+
+func (s *MethodTestSuite) TestAPIKey() {
+	s.Run("DeleteAPIKeyByID", s.Subtest(func(db database.Store, check *expects) {
+		key, _ := dbgen.APIKey(s.T(), db, database.APIKey{})
+		check.Args(key.ID).Asserts(key, rbac.ActionDelete).Returns()
+	}))
+	s.Run("GetAPIKeyByID", s.Subtest(func(db database.Store, check *expects) {
+		key, _ := dbgen.APIKey(s.T(), db, database.APIKey{})
+		check.Args(key.ID).Asserts(key, rbac.ActionRead).Returns(key)
+	}))
+	s.Run("GetAPIKeyByName", s.Subtest(func(db database.Store, check *expects) {
+		key, _ := dbgen.APIKey(s.T(), db, database.APIKey{
+			TokenName: "marge-cat",
+			LoginType: database.LoginTypeToken,
+		})
+		check.Args(database.GetAPIKeyByNameParams{
+			TokenName: key.TokenName,
+			UserID:    key.UserID,
+		}).Asserts(key, rbac.ActionRead).Returns(key)
+	}))
+	s.Run("GetAPIKeysByLoginType", s.Subtest(func(db database.Store, check *expects) {
+		a, _ := dbgen.APIKey(s.T(), db, database.APIKey{LoginType: database.LoginTypePassword})
+		b, _ := dbgen.APIKey(s.T(), db, database.APIKey{LoginType: database.LoginTypePassword})
+		_, _ = dbgen.APIKey(s.T(), db, database.APIKey{LoginType: database.LoginTypeGithub})
+		check.Args(database.LoginTypePassword).
+			Asserts(a, rbac.ActionRead, b, rbac.ActionRead).
+			Returns(slice.New(a, b))
+	}))
```

Comment:
Would be really nice to generate these too. Your call on how difficult that is though.

Maybe I could make the test failure just print out the test to copy/paste. :thinking: 

happy to brainstorm this

Reply:
I think we can do that post-merge or something... right now it's not a massive pain-point for me, so I think we can punt it.